src/Node/Render.php

```diff
@@ -50,6 +51,35 @@ public function __construct(Twig_Node_Expression $expr, Twig_Node_Expression $va
     parent::__construct($expr, $variables, $only, $ignoreMissing, $lineno);
   }
 
+  protected function addGetTemplate(Twig_Compiler $compiler) {
+    $defaults = $this->getComponentDefaults($this->component);
+    $compiler->raw('$defaults = ')->repr($defaults)->raw(';');
+    $compiler->write(
+      'foreach ($context as $context_key => $context_value) {',
+        'if ($context_value instanceof \Drupal\Core\Template\Attribute) {',
+          'foreach ($defaults[\'attr\'] as $name => $value) {',
+            'if (!isset($context_value[$name])) {',
+              '$context_value[$name] = $value;',
+            '}',
+            'elseif ($name === \'class\') {',
+              '$context_value->addClass($value);',
+            '}',
+          '}',
+        '}',
+      '}'
+    );
```

Comment:
I'd rather use a nowdoc for this to make it more readable: http://php.net/manual/en/language.types.string.php#language.types.string.syntax.nowdoc